ext/ffi/00_ffi.js

```diff
@@ -326,6 +334,33 @@
             },
           );
         }
+
+        if (needsUnpacking && !isNonBlocking) {
+          const call = this.symbols[symbol];
+          const parameters = symbols[symbol].parameters;
+          const vi = new Int32Array(2);
+          const vui = new Uint32Array(vi.buffer);
+          const b = new BigInt64Array(vi.buffer);
+
+          const params = parameters.map((_, index) => `p${index}`);
+          // Make sure V8 has no excuse to not optimize this function.
+          this.symbols[symbol] = new Function(
+            "vi",
+            "vui",
+            "b",
+            "call",
+            `return function (${params.join(", ")}) {
+            call(${params.join(", ")}${parameters.length > 0 ? ", " : ""}vi);
```

Comment:
The `${parameters.length > 0 ? ", " : ""}vi` may be avoided by pushing `"vi"` to `params` after mapping?